### PR TITLE
refactor(gas): Restructure gas consumption

### PR DESCRIPTION
**Pull request by itsdevbear**
attempt at fixing potential gas issues in consensus: https://github.com/cosmos/cosmos-sdk/issues/12583

### CODE REVIEW

**Review by ocnc — APPROVED** [review]

File: cosmos/x/evm/keeper/processor.go [hunk 1]
```diff
@@ -22,45 +22,51 @@ package keeper
...
-	sCtx.GasMeter().RefundGas(sCtx.GasMeter().GasConsumed(),
+	gasMeter.RefundGas(gasMeter.GasConsumed(),
 		"reset gas meter prior to ethereum state transition")
 
 	// Process the transaction and return the EVM's execution result.
-	execResult, err := k.miner.ProcessTransaction(ctx, tx)
+	receipt, err := k.miner.ProcessTransaction(ctx, tx)
 	if err != nil {
 		return nil, err
 	}
```
Comment (calbera) [inline cosmos/x/evm/keeper/processor.go:46]: I think this case should also log the error

File: cosmos/x/evm/keeper/processor.go [hunk 2]
```diff
@@ -22,45 +22,51 @@ package keeper
...
 
-	// We don't want the cosmos transaction to be marked as failed if the EVM reverts. But
-	// its not the worst idea to log the error.
-	if execResult.Err != nil {
-		k.Logger(sdk.UnwrapSDKContext(ctx)).Error(
-			"evm execution",
-			"tx_hash", tx.Hash(),
-			"error", execResult.Err,
-			"gas_consumed", sCtx.GasMeter().GasConsumed(),
-		)
```
Comment (calbera) [inline cosmos/x/evm/keeper/processor.go:55]: this case as well

Comment (itsdevbear) [reply cosmos/x/evm/keeper/processor.go:55]: nah let's simplify we shouldn't care, this approach reduces the geth fork by about 100 lines

File: cosmos/x/evm/keeper/processor.go (hunk 1 above)
Comment (itsdevbear) [reply cosmos/x/evm/keeper/processor.go:46]: yeah sure

**Review by calbera — APPROVED** [review]
Log the error case then lgtm

**Comment by taeguk** [comment]
@itsdevbear @calbera you guys saw consensus failure (like https://github.com/cosmos/cosmos-sdk/issues/12583) and fixed it? Or just fixed it for potential problems?

**Pull request merged**